### PR TITLE
Refactored some things, improved flair reports

### PR DESCRIPTION
**Pull request by not-an-aardvark**

### CODE REVIEW

File: api/controllers/FlairController.js [hunk 1]
```diff
@@ -132,88 +132,81 @@ module.exports = {
...
-        res.status(500).json({error: "Unknown error"});
-      }
-      var now = moment();
-      if (events.length) {
-        var then = moment(events[0].createdAt);
-        then.add(2, 'minutes');
-        if (then.isAfter(now)) {
-          return res.status(400).json({error: "You set your flair too recently, please try again in a few minutes."});
-        }
+    var events = await Event.find(appData);
```
Comment (YaManicKill) [inline api/controllers/FlairController.js]: Wait, waterline supports promises now? This. Changes. Everything.

File: api/controllers/FlairController.js [hunk 2]
```diff
@@ -132,88 +132,81 @@ module.exports = {
...
-        }
+    var flagged = _.reject(flairs.fcs, Flairs.validFC);
+    var ipAddress = req.headers['x-forwarded-for'] || req.ip;
+    // Get IP matches with banned users
+    var banned_alts = await Event.find({content: {contains: ipAddress}, user: {not: req.user.name}}).then(function (eventResults) {
+      return User.find({name: _.uniq(_.map(eventResults, 'user')), banned: true});
+    });
+    // Get friend codes that are similar (have a low edit distance) to banned friend codes
+    var similar_banned_fcs = _.flatten(await* flairs.fcs.map(Flairs.getSimilarBannedFCs));
+    console.log(similar_banned_fcs);
```
Comment (YaManicKill) [inline api/controllers/FlairController.js]: Did you mean to leave this in?

Comment (not-an-aardvark) [inline api/controllers/FlairController.js]: No, it was replaced by the _.reject line.

File: api/controllers/FlairController.js (hunk 1 above)
Comment (not-an-aardvark) [inline api/controllers/FlairController.js]: According to [the docs](http://sailsjs.org/documentation/reference/waterline-orm/queries#?promises) it supports Bluebird promises, not native promises. Apparently it works with await though.

**Pull request merged**